### PR TITLE
Cleaned up path creation and drupal checks.

### PR DESCRIPTION
**Pull request by penyaskito**

### CODE REVIEW

**Comment by penyaskito** [comment]
Impressive! This is green locally :(

File: src/Discovery/PathFinderCore.php [hunk 1]
```diff
@@ -17,19 +17,18 @@ class PathFinderCore extends PathFinderBase implements PathFinderInterface {
...
-      $directory = new \DirectoryIterator($path_info['dirname']);
+    else {
+      $directory = new \DirectoryIterator(dirname($seed));
     }
     // Starting at the directory containing the seed path, we go one directory
     // up and up and up until we reach the Drupal root.
     do {
       if ($this->isDrupalRoot($directory)) {
-        return $this->cleanDirPath($directory->getPathName()) . $this->path;
+        return dirname($this->cleanDirPath($directory->getPathName())) . $this->path;
```
Comment (e0ipso) [inline src/Discovery/PathFinderCore.php]: What is this for?

File: src/Discovery/PathFinderCore.php [hunk 2]
```diff
@@ -54,15 +53,14 @@ protected function isDrupalRoot(\DirectoryIterator $directory) {
     // operator.
     $d = clone $directory;
     // Check if there is a COPYRIGHT.txt file in the directory.
-    foreach ($d as $item) {
-      if (!$item->isFile() || $item->getFilename() != 'COPYRIGHT.txt') {
-        continue;
-      }
+    $copyrightPath = $d->getPath() . DIRECTORY_SEPARATOR . 'COPYRIGHT.txt';
+    $check = file_exists($copyrightPath);
+    if ($check) {
```
Comment (e0ipso) [inline src/Discovery/PathFinderCore.php]: Maybe:

``` php
if (!$check = file_exists($copyrightPath)) {
  return FALSE;
}
$line = fgets(fopen($copyrightPath, 'r'));
return strpos($line, 'All Drupal code is Copyright') === 0;
```

File: src/Discovery/PathFinderCore.php (hunk 1 above)
Comment (e0ipso) [inline src/Discovery/PathFinderCore.php]: Oh! I see, you want to replace `cleanDirPath` by `dirname`.

Comment (e0ipso) [inline src/Discovery/PathFinderCore.php]: Not sure if that is a good idea though. The code behaves differently when the path does not end in `/.`.

File: tests/src/Discovery/PathFinderCoreTest.php [hunk 3]
```diff
@@ -41,8 +41,26 @@ public function test_find() {
...
+
+  /**
+   * Tests that ::find() works properly.
+   *
+   * @expectedException \Drupal\Composer\ClassLoader\ClassLoaderException
+   *
+   * @covers ::getParentDirectory()
+   */
+  public function test_getParentDirectory__root() {
+
```
Comment (e0ipso) [inline tests/src/Discovery/PathFinderCoreTest.php]: Can we remove these extra new lines.

File: tests/src/Discovery/PathFinderCoreTest.php [hunk 4]
```diff
@@ -41,8 +41,26 @@ public function test_find() {
...
+
+    // 2. Test seed not in Drupal root.
```
Comment (e0ipso) [inline tests/src/Discovery/PathFinderCoreTest.php]: This should be :three:

File: tests/src/Discovery/PathFinderCoreTest.php [hunk 5]
```diff
@@ -41,8 +41,26 @@ public function test_find() {
    */
   public function test_getParentDirectory__noParent() {
```
Comment (e0ipso) [inline tests/src/Discovery/PathFinderCoreTest.php:42]: Why is this test not covering like `test_getParentDirectory__root` :question:

Comment (penyaskito) [inline tests/src/Discovery/PathFinderCoreTest.php:42]: OK, the issue is that test_find__noContrib was throwing the exception, but in PathCoreFinder. Editing existing test instead with a mock, so we are back to 100% coverage.

File: tests/src/Discovery/PathFinderCoreTest.php (hunk 4 above)
Comment (penyaskito) [inline tests/src/Discovery/PathFinderCoreTest.php]: :arrow_up:

**Pull request merged**